modules/indexing/src/test/java/org/apache/ignite/cache/query/IndexQueryPartitionTest.java

```diff
@@ -0,0 +1,209 @@
+        }
+
+        assertTrue(expRes.isEmpty());
+
+        // Send request to single node only.
+        assertEquals(1, TestRecordingCommunicationSpi.spi(grid(0)).recordedMessages(true).size());
+    }
+
+    /** */
+    @Test
+    public void testSetNullNotAffect() {
+        try (IgniteDataStreamer<Integer, Person> dataStreamer = grid(0).dataStreamer("CACHE")) {
+            Random rnd = new Random();
+
+            for (int i = 0; i < 10_000; i++)
+                dataStreamer.addData(i, new Person(rnd.nextInt()));
+        }
+
+        IndexQuery<Integer, Person> qry = new IndexQuery<Integer, Person>(Person.class)
+            .setPartition(0);
+
+        assertTrue(grid(0).cache("CACHE").query(qry).getAll().size() < 10_000);
+
+        qry = new IndexQuery<Integer, Person>(Person.class)
+            .setPartition(null);
+
+        assertTrue(grid(0).cache("CACHE").query(qry).getAll().size() == 10_000);
+    }
+
+    /** */
+    @Test
+    public void testLocalWithPartition() {
+        try (IgniteDataStreamer<Integer, Person> dataStreamer = grid(0).dataStreamer("CACHE")) {
+            Random rnd = new Random();
+
+            for (int i = 0; i < 10_000; i++)
+                dataStreamer.addData(i, new Person(rnd.nextInt()));
+        }
+
+        IndexQuery<Integer, Person> qry = new IndexQuery<Integer, Person>(Person.class)
+            .setPartition(0);
+
+        qry.setLocal(true);
+
+        ClusterNode node = grid(0).affinity("CACHE").mapPartitionToNode(0);
+
+        for (Ignite ign: G.allGrids()) {
+            if (node.equals(((IgniteEx)ign).localNode()))
+                assertTrue(!ign.cache("CACHE").query(qry).getAll().isEmpty());
+            else {
+                GridTestUtils.assertThrows(null, () -> ign.cache("CACHE").query(qry).getAll(),
+                    IgniteException.class,
+                    "Cluster group is empty.");
+            }
+        }
+    }
+
+    /** */
+    @Test
+    public void testNegativePartitionFails() {
+        GridTestUtils.assertThrows(null, () -> new IndexQuery<Integer, Person>(Person.class).setPartition(-1),
+            IllegalArgumentException.class,
+            "Specified partition must be in the range [0, N) where N is partition number in the cache.");
+
+        GridTestUtils.assertThrows(null, () -> new IndexQuery<Integer, Person>(Person.class).setPartition(-23),
+            IllegalArgumentException.class,
+            "Specified partition must be in the range [0, N) where N is partition number in the cache.");
+
+        GridTestUtils.assertThrows(null, () -> {
+            IndexQuery qry = new IndexQuery<Integer, Person>(Person.class).setPartition(1000);
+
+            grid(0).cache("CACHE").query(qry);
+        },
+            IgniteException.class,
+            "Specified partition must be in the range [0, N) where N is partition number in the cache.");
+    }
+
+    /** */
+    private static class Person {
+        /** */
+        @GridToStringInclude
```

Comment:
Why is it required to exclude this field from a string representation of the object?

Reply:
It's included (GridToString**Include**) actually.